calendar_resource/models/calendar_event.py

```diff
@@ -0,0 +1,268 @@
+# Copyright 2017 Laslabs Inc.
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from os import linesep
+from datetime import datetime, time, timedelta
+from dateutil.rrule import rrule, DAILY
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class CalendarEvent(models.Model):
+
+    _inherit = 'calendar.event'
+
+    resource_ids = fields.Many2many(
+        string='Resources',
+        comodel_name='resource.resource',
+    )
+
+    @api.model
+    def _format_datetime_intervals_to_str(self, intervals):
+        """ Converts intervals to string values.
+
+        Example:
+
+            .. code-block python
+
+            self._format_datetime_intervals_to_str(
+                intervals=[
+                    (
+                        datetime(2017, 6, 28, 17, 0, 0),
+                        datetime(2017, 6, 29, 8, 0, 0),
+                    ),
+                    (
+                        datetime(2017, 6, 29, 12, 0, 0),
+                        datetime(2017, 6, 29, 13, 0, 0),
+                    ),
+                ],
+            )
+
+            The above example will output the following string,
+            depending on date and time format, as well as timezone.
+
+            \n06/28/2017 at 17:00:00 To\n
+            06/29/2017 at 08:00:00 (UTC)\n
+
+            \n06/29/2017 at 12:00:00 To\n
+            06/29/2017 at 13:00:00 (UTC)\n
+
+        Returns:
+
+            (str): string formatted for front-end
+            display purposes, taking into account language, timezone, as
+            well as date and time formats.
+
+        """
+        datetimes = []
+        for interval in intervals:
+            if not isinstance(interval[0], str):
+                interval = (
+                    fields.Datetime.to_string(interval[0]),
+                    fields.Datetime.to_string(interval[1]),
+                )
+            args = {
+                'start': interval[0],
+                'stop': interval[1],
+                'zallday': False,
+                'zduration': 24,
+            }
+            datetimes.append(self._get_display_time(**args))
+        return (2 * linesep).join(datetimes)
+
+    @api.multi
+    def _event_in_past(self):
+        self.ensure_one()
+        stop_datetime = fields.Datetime.from_string(self.stop)
+        now_datetime = datetime.now()
+        return stop_datetime < now_datetime
+
+    @api.multi
+    @api.constrains('resource_ids', 'start', 'stop')
+    def _check_resource_ids_double_book(self):
+
+        for record in self:
+
+            if record._event_in_past():
+                continue
+
+            resources = record.resource_ids.filtered(
+                lambda s: s.allow_double_book is False
+            )
+
+            if not any(resources):
+                continue
+
+            overlaps = self.env['calendar.event'].search([
+                ('id', '!=', record.id),
```

Comment:
```suggestion
                ('id', '!=', record.id),
                ('resource_ids', '!=', False),
```

Reply:
Tanks, that's a  good extra filter condition